softlayer/resource_softlayer_scale_group.go

```diff
@@ -309,6 +301,36 @@ func resourceSoftLayerScaleGroupCreate(d *schema.ResourceData, meta interface{})
 	return resourceSoftLayerScaleGroupRead(d, meta)
 }
 
+func buildLoadBalancers(d *schema.ResourceData) ([]datatypes.Scale_LoadBalancer, error) {
+	isLoadBalancerEmpty := true
+	loadBalancers := []datatypes.Scale_LoadBalancer{{}}
+
+	if virtualServerId, ok := d.GetOk("virtual_server_id"); ok {
+		isLoadBalancerEmpty = false
+		loadBalancers[0].VirtualServerId = sl.Int(virtualServerId.(int))
+	}
+
+	if healthCheck, ok := d.GetOk("health_check"); ok {
+		isLoadBalancerEmpty = false
+		healthCheckOpts, err := buildHealthCheckFromResourceData(healthCheck.(map[string]interface{}))
+		if err != nil {
+			return []datatypes.Scale_LoadBalancer{}, fmt.Errorf("Error while parsing health check options: %s", err)
+		}
+		loadBalancers[0].HealthCheck = &healthCheckOpts
+	}
+
+	if port, ok := d.GetOk("port"); ok {
+		isLoadBalancerEmpty = false
+		loadBalancers[0].Port = sl.Int(port.(int))
+	}
+
+	if isLoadBalancerEmpty {
+		return []datatypes.Scale_LoadBalancer{}, nil
+	} else {
+		return loadBalancers, nil
+	}
+}
```

Comment:
`len(loadBalancers)` is `1` because `loadBalancers := []datatypes.Scale_LoadBalancer{{}}`